### PR TITLE
build(ci): Tweak usage of `actions/cache`

### PR DESCRIPTION
**Pull request by billyvg**
* Remove usage of `restore-keys` as it can continuously grow the cache size and cause restoring cache to be slow. The trade-off is that cache misses will be slower. This is acceptable as these dependencies should not change often.
* Requires further investigation, but I believe `actions/cache@v2` has a perf regression when GHA logging is enabled. See some tests here: [v1: 14s](https://github.com/getsentry/sentry/runs/3235769855#step:7:1) vs [v2 - 36s](https://github.com/getsentry/sentry/pull/27917/checks?check_run_id=3235884831#step:7:1)

Other things to note in regards to `yarn install`:
* [bahmutov/npm-install action](https://github.com/bahmutov/npm-install) can be useful as it abstracts the caching (less yaml boilerplate)
  * the new `action/setup-node` action from GH also does this (but we use volta to setup node)
  * [pnpm is fast af](https://github.com/getsentry/sentry/runs/3235769953)

The combination of starting with a new cache (our previous cache snowballed from around 180MB to nearly 600MB) and switching from `v2` -> `v1` should bring an improvement from around 90 seconds down to 15-20 seconds.

### CODE REVIEW

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -40,13 +40,10 @@ jobs:
         id: yarn-cache-dir-path
         run: echo "::set-output name=dir::$(yarn cache dir)"
 
-      - uses: actions/cache@v2
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
```
Comment (billyvg) [inline .github/workflows/acceptance.yml:44]: We never used this

**Comment by joshuarli** [comment]
> perf regression when logging is enabled

classic

**Comment by billyvg** [comment]
Looks like they list the contents of the cache archive if debugging is enabled: https://github.com/actions/toolkit/blob/main/packages/cache/src/cache.ts#L103-L105

I've created an issue [here](https://github.com/actions/toolkit/issues/872)

File: .github/workflows/acceptance.yml [hunk 2]
```diff
@@ -40,13 +40,10 @@ jobs:
...
+      - uses: actions/cache@v1 # We are explicitly using v1 due to perf reasons
```
Comment (armenzg) [inline .github/workflows/acceptance.yml:43]: If we didn't use the debug logging we could be using v2 and have better perf. No?
image

File: .github/workflows/acceptance.yml [hunk 3]
```diff
@@ -40,13 +40,10 @@ jobs:
...
         with:
           path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock', 'api-docs/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
+          key: ${{ runner.os }}-v2-yarn-${{ hashFiles('yarn.lock', 'api-docs/yarn.lock') }}
```
Comment (armenzg) [inline .github/workflows/acceptance.yml:46]: When you mention `v2`, are you referring to a 2nd version of our key naming?

File: .github/workflows/acceptance.yml (hunk 2 above)
Comment (billyvg) [reply .github/workflows/acceptance.yml:43]: Yes, but I'd prefer to have debug logging on

File: .github/workflows/acceptance.yml (hunk 3 above)
Comment (billyvg) [reply .github/workflows/acceptance.yml:46]: Yeah this is to bust the cache

**Review by armenzg — APPROVED** [review]

**Pull request merged**